net/pfSense-pkg-freeradius2/files/usr/local/pkg/freeradius.inc

```diff
@@ -4409,4 +4415,834 @@ EOD;
+		$input_errors[] = "Do not use special characters in the 'Description' field; only /^[a-zA-Z0-9 _,.;:+=()-]*$/ allowed.";
+	}
+
+	/*
+	 * TODO: Check that the configured port is unique for the selected Interface Type/IP address.
+	 */
+
+}
+
+/* General Settings input validation */
+function freeradius_validate_settings($post, &$input_errors) {
+
+	// Maximum Requests Tracked
+	if ($post['varsettingsmaxrequests'] != '' && !is_numericint($post['varsettingsmaxrequests'])) {
+		$input_errors[] = "The 'Maximum Requests Tracked' field must contain an integer value.";
+	}
+
+	// Maximum Request Timeout
+	if ($post['varsettingsmaxrequesttime'] != '' && !is_numericint($post['varsettingsmaxrequesttime'])) {
+		$input_errors[] = "The 'Maximum Request Timeout' field must contain an integer value.";
+	}
+
+	// Cleanup Delay
+	if ($post['varsettingscleanupdelay'] != '' && !is_numericint($post['varsettingscleanupdelay'])) {
+		$input_errors[] = "The 'Cleanup Delay' field must contain an integer value.";
+	}
+
+	// Maximum Number of Attributes
+	if ($post['varsettingsmaxattributes'] != '' && !is_numericint($post['varsettingsmaxattributes'])) {
+		$input_errors[] = "The 'Maximum Number of Attributes' field must contain an integer value.";
+	}
+
+	// Access-Reject Delay
+	if ($post['varsettingsrejectdelay'] != '' && !is_numericint($post['varsettingsrejectdelay'])) {
+		$input_errors[] = "The 'Access-Reject Delay' field must contain an integer value.";
+	}
+
+	// Number of Threads After Start
+	if ($post['varsettingsstartservers'] != '' && !is_numericint($post['varsettingsstartservers'])) {
+		$input_errors[] = "The 'Number of Threads After Start' field must contain an integer value.";
+	}
+
+	// Maximum Number of Threads
+	if ($post['varsettingsmaxservers'] != '' && !is_numericint($post['varsettingsmaxservers'])) {
+		$input_errors[] = "The 'Maximum Number of Threads' field must contain an integer value.";
+	}
+
+	// Minimum Spare Servers
+	if ($post['varsettingsminspareservers'] != '' && !is_numericint($post['varsettingsminspareservers'])) {
+		$input_errors[] = "The 'Minimum Spare Servers' field must contain an integer value.";
+	}
+
+	// Maximum Spare Servers
+	if ($post['varsettingsmaxspareservers'] != '' && !is_numericint($post['varsettingsmaxspareservers'])) {
+		$input_errors[] = "The 'Maximum Spare Servers' field must contain an integer value.";
+	}
+
+	// Server Packet Queue Size
+	if ($post['varsettingsmaxqueuesize'] != '' && !is_numericint($post['varsettingsmaxqueuesize'])) {
+		$input_errors[] = "The 'Server Packet Queue Size' field must contain an integer value.";
+	}
+
+	// Maximum Requests Per Server
+	if ($post['varsettingsmaxrequestsperserver'] != '' && !is_numericint($post['varsettingsmaxrequestsperserver'])) {
+		$input_errors[] = "The 'Maximum Requests Per Server' field must contain an integer value.";
+	}
+
+	// OTP Lifetime
+	if ($post['varsettingsmotpenable'] == 'on') {
+		if ($post['varsettingsmotptimespan'] === 0) {
```

Comment:
If you are going to test using ===, you should cast to (int) or similar on the variable since it would be a string.